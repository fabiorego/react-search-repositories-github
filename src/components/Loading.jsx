import React from 'react';

export default function Loading(props) {
  const loading = props.display;
  return (
    <div className="" style={{ display: loading }}>
      <div className="">
        <div id="loading" class="loading-container">
          <div class="d-flex justify-content-center align-items-center loader">
            <div class="spinner-border text-primary" role="status">
              <span class="sr-only">Loading...</span>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
