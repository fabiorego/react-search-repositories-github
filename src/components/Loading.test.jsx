import React from 'react';
import Loading from './Loading';
import renderer from 'react-test-renderer';

let findTextElement = function(tree, element) {
  return true;
};

test('Loading snapshot', () => {
  const component = renderer.create(<Loading></Loading>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Find Moment element', () => {
  let tree = renderer.create(<Loading />).toJSON();
  expect(findTextElement(tree, 'Moment')).toBeDefined();
});
