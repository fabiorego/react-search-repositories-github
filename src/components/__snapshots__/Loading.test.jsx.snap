// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshot 1`] = `
<div
  className="loading-content"
  style={
    Object {
      "display": undefined,
    }
  }
>
  <div
    className="loading-image"
  >
    <i
      className="fas fa-spinner spinnerLoading"
    />
  </div>
  <div
    className="loading-text"
  >
    One Moment
  </div>
</div>
`;
